styles/mixins/spacing.js

```diff
@@ -8,27 +8,47 @@ module.exports = function (config) {
 
     for (const breakpoint of spacingBreakpoints) {
       styles[breakpoint] = {};
-      styles[breakpoint][`${prop}-${position}`] = config.spacing[breakpoint][size];
+      if (typeof position === 'string') {
+        styles[breakpoint][`${prop}-${position}`] = config.spacing[breakpoint][size];
+      } else {
+        position.forEach((pos) => {
+          styles[breakpoint][`${prop}-${pos}`] = config.spacing[breakpoint][size];
+        });
+      }
     }
 
     return breakpointHelper(styles, config);
   };
 
+
   return function (mixin, prop, position, size) {
     if (prop && position && size) {
       return spacingMixin(prop, position, size);
     }
     const styles = {};
     const properties = ['padding', 'margin'];
-    const positions = ['top', 'left', 'right', 'bottom'];
+    const positions = ['top', 'left', 'right', 'bottom', 'xaxis', 'yaxis'];
     const sizes = Object.keys(config.spacing.default);
+    const axials = {
+      xaxis: ['left', 'right'],
+      yaxis: ['top', 'bottom']
+    };
     properties.forEach((property) => {
-      positions.forEach((position) => {
+      let addedAllProperty = false;
+      positions.forEach((positionString) => {
         sizes.forEach((size) => {
-          styles[`.${property[0]}-${position[0]}-${size}`] = spacingMixin(property, position, size);
+          if (!addedAllProperty) {
+            addedAllProperty = true;
+            styles[`.${property[0]}-${size}`] = spacingMixin(property, positionString, size);
+          }
+          if (Object.keys(axials).indexOf(positionString) > -1) {
+            styles[`.${property}-${positionString[0]}-${size}`] = spacingMixin(property, axials[positionString], size);
```

Comment:
instead of changing the other function, have this add the two elements to object